From dfe27f623765b20bdaff2306109f89baca67d0cf Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 19 Feb 2010 03:29:36 +0100
Subject: [PATCH 45/62] shm: don't complain about missing SHM segments

If two clients try to cleanup the SHM directory at the same time, they
might want to open and then delete the same segment at the same time, in
which case one client might win, the other one lose. In this case, don't
warn about ENOENT.
---
 src/pulsecore/shm.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/pulsecore/shm.c b/src/pulsecore/shm.c
index fbf777a..1e31d54 100644
--- a/src/pulsecore/shm.c
+++ b/src/pulsecore/shm.c
@@ -286,7 +286,7 @@ int pa_shm_attach_ro(pa_shm *m, unsigned id) {
     segment_name(fn, sizeof(fn), m->id = id);
 
     if ((fd = shm_open(fn, O_RDONLY, 0)) < 0) {
-        if (errno != EACCES)
+        if (errno != EACCES && errno != ENOENT)
             pa_log("shm_open() failed: %s", pa_cstrerror(errno));
         goto fail;
     }
-- 
1.7.0

